upgrade.sh

```diff
@@ -67,6 +67,67 @@ undeploy -n gatekeeper-system gatekeeper-constraints
 undeploy -n gatekeeper-system gatekeeper-policy-library
 undeploy -n gatekeeper-system gatekeeper
 
+cluster_role=$(kubectl get clusterrole | grep gatekeeper | awk '{print $1}')
```

Comment:
I see two `clusterroles` for gatekeeper, `gatekeeper-manager-role` and `gatekeeper-policy-manager-crd-view`. Both of these are managed by Helm, and thus, should be properly removed when you remove the owning chart.

Looking at mug, these charts are all still installed:

```
ncn-m001:~ # helm ls -n gatekeeper-system
NAME                     	NAMESPACE        	REVISION	UPDATED                                	STATUS  	CHART                          	APP VERSION
gatekeeper               	gatekeeper-system	2       	2022-09-15 18:13:29.740362023 +0000 UTC	deployed	gatekeeper-1.5.2               	v3.1.1     
gatekeeper-constraints   	gatekeeper-system	2       	2022-09-15 18:13:43.642197705 +0000 UTC	deployed	gatekeeper-constraints-0.5.0   	1.16.0     
gatekeeper-policy-library	gatekeeper-system	2       	2022-09-15 18:13:37.426899242 +0000 UTC	deployed	gatekeeper-policy-library-0.5.0	1.16.0     
gatekeeper-policy-manager	gatekeeper-system	2       	2022-09-15 18:18:19.455606823 +0000 UTC	deployed	gatekeeper-policy-manager-1.3.0	0.4.0 
```

And, for example, the `gatekeeper-manager-role` is managed by the gatekeeper helm release. 

So, have you actually verified that these policies still exist after deinstall of the Helm charts?

Reply:
Yes, If we delete/shutdown all the pods gracefully and uninstall will clean these roles. uninstall called during upgrade, We will check this scenario in any 1.4 upgrade system.

Reply:
Implemented the review comment. Tested on mug and attached the latest logs.